akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesNewExtractorSpec.scala

```diff
@@ -0,0 +1,297 @@
+  }
+
+  def startShardingWithExtractor2(): Unit = {
+    ClusterSharding(system).start(
+      typeName = typeName,
+      entityProps = ClusterShardingRememberEntitiesNewExtractorSpec.props(Some(testActor)),
+      settings = ClusterShardingSettings(system).withRememberEntities(true).withRole("sharding"),
+      extractEntityId = extractEntityId,
+      extractShardId = extractShardId2)
+  }
+
+  lazy val region = ClusterSharding(system).shardRegion(typeName)
+
+  def isDdataMode: Boolean = mode == ClusterShardingSettings.StateStoreModeDData
+
+  s"Cluster with min-nr-of-members using sharding ($mode)" must {
+
+    if (!isDdataMode) {
+      "setup shared journal" in {
+        // start the Persistence extension
+        Persistence(system)
+        runOn(first) {
+          system.actorOf(Props[SharedLeveldbStore], "store")
+        }
+        enterBarrier("persistence-started")
+
+        runOn(second, third, fourth, fifth) {
+          system.actorSelection(node(first) / "user" / "store") ! Identify(None)
+          val sharedStore = expectMsgType[ActorIdentity](10.seconds).ref.get
+          SharedLeveldbJournal.setStore(sharedStore, system)
+        }
+
+        enterBarrier("after-1")
+      }
+    }
+
+    "start up first cluster and sharding" in within(15.seconds) {
+      join(first, first)
+      join(second, first)
+      join(third, first)
+
+      runOn(first, second, third) {
+        within(remaining) {
+          awaitAssert {
+            cluster.state.members.count(_.status == MemberStatus.Up) should ===(3)
+          }
+        }
+      }
+      runOn(second, third) {
+        startShardingWithExtractor1()
+      }
+      enterBarrier("first-cluster-up")
+
+      runOn(second, third) {
+        // one entity for each shard id
+        (1 to 10).foreach { n ⇒
+          region ! n
+          expectMsg(n)
+        }
+      }
+      enterBarrier("first-cluster-entities-up")
+    }
+
+    "shutdown sharding nodes" in within(30.seconds) {
+      runOn(first) {
+        testConductor.exit(second, 0).await
+        testConductor.exit(third, 0).await
+      }
+      runOn(first) {
+        within(remaining) {
+          awaitAssert {
+            cluster.state.members.count(_.status == MemberStatus.Up) should ===(1)
+          }
+        }
+      }
+      enterBarrier("first-sharding-cluster-stopped")
+    }
+
+    "start new nodes with different extractor" in within(15.seconds) {
+
+      // start it with a new shard id extractor, which will put the entities
+      // on different shards
+
+      join(fourth, first)
+      join(fifth, first)
+      runOn(first) {
+        within(remaining) {
+          awaitAssert {
+            cluster.state.members.count(_.status == MemberStatus.Up) should ===(3)
+          }
+        }
+      }
+      runOn(fourth, fifth) {
+        startShardingWithExtractor2()
+      }
+
+      // TODO how do we know that the shards has started??
+      Thread.sleep(7000)
```

Comment:
That requires there to be a shard on both nodes, which does not happen all the time, for some reason that I didn't yet figure out.